internal/controller/clustersecurityintentbinding_controller.go

```diff
@@ -35,71 +39,184 @@ func (r *ClusterSecurityIntentBindingReconciler) Reconcile(ctx context.Context,
 	logger := log.FromContext(ctx)
 
 	csib := &v1.ClusterSecurityIntentBinding{}
-	err := r.Get(ctx, types.NamespacedName{Name: req.Name}, csib)
+	err := r.Get(ctx, req.NamespacedName, csib)
 	if err != nil {
-		if errors.IsNotFound(err) {
+		if apierrors.IsNotFound(err) {
 			logger.Info("ClusterSecurityIntentBinding not found. Ignoring since object must be deleted")
-			return ctrl.Result{}, nil
+			logger.Info("ClusterNimbusPolicy deleted due to ClusterSecurityIntentBinding deletion",
```

Comment:
I know that we are not deleting the cluster Nimbus policy since the garbage collector will do the job. However, that will take some time - depending on when the garbage collector runs. Also, we should depend on the garbage collector if we are unable to delete ourselves (for example - because the controller has crashed).

Is there any harm in deleting the cluster Nimbus policy here ?

Reply:
Any option is fine here - either explicitly deleting the object or leaving it to garbage collection..